### DISCUSSION TITLE
Procedural state-machine modeling languages

### DISCUSSION THREAD

**Discussion by shonfeder** [informalsystems/quint#202] [Ideas]
There seems to be broad agreement around the idea that at some point we'll want to support a more procedural-flavored language for specifying systems.

This is a place to gather general brainstorming and ideas around languages or language features we should target, support, or develop.

**Comment by thpani** [comment]
> There seems to be broad agreement around the idea that at some point we'll want to support a more procedural-flavored language for specifying systems.

I thought Quint was going to be that language.
My main concern is that Quint is in a weird limbo state, where it has some procedural flavor (e.g., assignment via `<-`), but still lives in the world of logic when it comes to encoding transitions (`&` and `|`).

I think we should be clear on what users need to get their job done, before spending a lot of effort on multiple languages.

**Comment by konnov** [comment]
I am not sure that procedural flavor is in in our plans in the near future. Quite the opposite, the functional layer of Quint started to prove its value. I am closing this discussion. Maybe we will come back to it in a few quarters from now.
